pkg/api/v1alpha1/tinkerbellmachineconfig.go

```diff
@@ -88,3 +88,36 @@ func WithTemplateRef(ref ProviderRefAccessor) TinkerbellMachineConfigGenerateOpt
 		}
 	}
 }
+
+func validateTinkerbellMachineConfig(config *TinkerbellMachineConfig) error {
+	if err := validateObjectMeta(config.ObjectMeta); err != nil {
+		return fmt.Errorf("TinkerbellMachineConfig: %v", err)
+	}
+
+	if len(config.Spec.HardwareSelector) == 0 {
+		return fmt.Errorf("TinkerbellMachineConfig: missing spec.hardwareSelector: %v", config.Name)
```

Comment:
```suggestion
		return fmt.Errorf("TinkerbellMachineConfig: missing spec.hardwareSelector: %s", config.Name)
```

Since `config.Name` is a string